scout/adapter/mongo/variant.py

```diff
@@ -658,16 +662,22 @@ def get_region_vcf(
                 variant_file = case_obj["vcf_files"].get("vcf_sv")
             elif category == "str":
                 variant_file = case_obj["vcf_files"].get("vcf_str")
+            elif category == "cancer":
+                variant_file = case_obj["vcf_files"].get("vcf_cancer")
         elif variant_type == "research":
             if category == "snv":
                 variant_file = case_obj["vcf_files"].get("vcf_snv_research")
             elif category == "sv":
                 variant_file = case_obj["vcf_files"].get("vcf_sv_research")
 
         if not variant_file:
-            raise SyntaxError("Vcf file does not seem to exist")
+            raise FileNotFoundError("Vcf file does not seem to exist")
+
+        try:
+            vcf_obj = VCF(variant_file)
```

Comment:
Why would we do that you mean? If we can not instantiate we want to exit and raise an exception

Reply:
I understood, but I think it would be nice to create the object before, outside the scope of the try

Reply:
Why you mean? We don't want to continue if the vcf_obj can not be created

Reply:
I agree, I don't want the function to continue either. What I'm try ti say is that if you have a VCF and the vcf_obj is created then you need it later in the function, outside the scope of the try, that's why you have to instantiate it before.

Looks here, second answer: https://stackoverflow.com/questions/25666853/how-to-make-a-variable-inside-a-try-except-block-public

Reply:
Sorry I still don't understand, perhaps this would be easier to talk about face to face:)
When `vcf_obj` is instantiated there are two options:

1. the instantiation fails. We then raise an exception which will abort. We don't need the object
1. the instantiation does not fail. Then we continue and have the object available. 

In the thread you link to they want to continue even if there is an exception which is a different case.
Is there anything that I'm missing?

Reply:
I don't think that the question in the thread is is about continuing even if there is an exception. I don't know, if a variable has to be used later in the function I always instantiate it before if or try blocks. I this case I think that the code will work anyway and since I'm tired of the discussion, whatever, keep it this way! 🙄